### ISSUE TITLE
strtok() can have one or two arguments

### ISSUE THREAD

**Issue by leofeyer** [vimeo/psalm#4874]
The following code generates a false-positive error:

https://psalm.dev/r/88c994db08

The PHP `strtok()` function can be called with either one or two arguments:

```php
$str = 'Foo,Bar,Baz';

echo strtok($str, ','); // Foo
echo strtok(',');       // Bar
echo strtok(',');       // Baz
```

See https://3v4l.org/WWFFt and https://www.php.net/manual/en/function.strtok.php

I assume this can be fixed by adjusting the [function stub](https://github.com/vimeo/psalm/blob/cc2933de351aa5c980928a088b252da12d2e5f17/stubs/CoreGenericFunctions.phpstub#L817), however, I was not able to make it work.

**Comment by weirdan** [comment]
I'd suggest to just drop the stub for `strtok()`.

1. It's not pure.
2. The flow annotation references invalid var name
3. There's probably no way to fix the flow annotation
3. It has effectively two signatures (defined in `dictionaries/CallMap.php`)
